cockroachdb/templates/statefulset.yaml

```diff
@@ -339,6 +347,9 @@ spec:
         {{- else }}
           emptyDir: {}
         {{- end }}
+        {{ with .Values.statefulset.volumes }}
```

Comment:
@udnay It is already is the way you have defined. `volumes` section is independent of `initContainers` section and can be used independently.  Both `initContainers` and `volumes` section are defined inside statefulset section on same hierarchy. 

Q: Would it make more sense to have a different top-level value for volume mounts inside the statefulset?
A: It is already at top level inside the statefulset section.

Q: I can imagine that users might want to add a volume not used in an initContainer, wdyt? 
A: It can already do that by defining `statefulset.volumes` section and not defining the `statefulset.initContainers` section.

Q: With the current setup a user could only ever define volumes that are also used in an init container.
A: No, it could define the independent volumes as well.

Reply:
I have added the test cases for all the scenarios as well

Reply:
Okay. Got it. You want to move the volumeMounts outside. Done.

Reply:
Yes, I think it is ok.  we don't want to over complicate things.